OpenRA.Mods.Common/Traits/Conditions/ExternalCondition.cs

```diff
@@ -32,18 +39,29 @@ public class ExternalConditionInfo : ITraitInfo, Requires<ConditionManagerInfo>
 		public object Create(ActorInitializer init) { return new ExternalCondition(init.Self, this); }
 	}
 
-	public class ExternalCondition
+	public class ExternalCondition : ITick, INotifyCreated
 	{
-		class TimedToken
+		struct TimedToken
 		{
-			public int Token;
-			public int Expires;
+			public readonly int Expires;
+			public readonly int Token;
+			public readonly object Source;
+
+			public TimedToken(int token, Actor self, object source, int duration)
+			{
+				Token = token;
+				Expires = self.World.WorldTick + duration;
+				Source = source;
+			}
 		}
 
 		public readonly ExternalConditionInfo Info;
 		readonly ConditionManager conditionManager;
 		readonly Dictionary<object, HashSet<int>> permanentTokens = new Dictionary<object, HashSet<int>>();
-		readonly Dictionary<object, HashSet<TimedToken>> timedTokens = new Dictionary<object, HashSet<TimedToken>>();
+		readonly List<TimedToken> timedTokens = new List<TimedToken>();
```

Comment:
Can you please add a comment here that `timedTokens` is manually sorted by expiry?